superset/charts/commands/data.py

```diff
@@ -82,6 +82,10 @@ def set_query_context(self, form_data: Dict[str, Any]) -> QueryContext:
         except ValidationError as error:
             raise error
 
+        for query in self._query_context.queries:
+            if query.row_limit and query.metrics and not query.orderby:
+                query.orderby = [(query.metrics[0], False)]
```

Comment:
While I understand the use case and agree with it from an end user perspective, always adding orderbys to queries can have a significant performance impact on queries referencing large tables. Going forward we want to make less implicit assumptions about ordering and let the viz' explicitly request ordering if it's needed. Therefore I feel this logic should be moved to the individual plugins rather than apply it on all queries.

Reply:
That sounds reasonable, however the behavior of returning unordered results in these cases seems to be a regression, as this wasn't happening in v0.37. Perhaps the viz components changed?

Reply:
If no one else has a problem with this, you can go ahead and close it and we'll patch it on our side.

Reply:
IIRC, word cloud was using the legacy API pre 0.37. Maybe that's why?

I added a [“sort by metric” option](https://github.com/apache-superset/superset-ui/pull/831) for Sankey not long ago (disabled by default), maybe you can take a similar approach?

Reply:
@bryanck originally this regression was caused by #11153 which removed the implicit orderby from chart data requests. I agree with the fix on a chart-by-chart basis, and definitely think it makes sense on word cloud. The problem with implicitly assuming an orderby clause is that it will cause unnecessary query overhead in cases where it is potentially not required, and sometimes cause unwanted side-effects. One side-effect that was noticed by this implicit ordering was on timeseries charts, where ordering by metric caused "scanning" from top to bottom if the limit was reached (=the observations with the lowest metric values didn't get through). In the case of a timeseries chart it feels more natural to order the results by series, not metrics, so as to make sure the request contains as many full series as possible.

Again I'd urge adding this fix to each chart separately, and also adding a sort control to charts where one might be needed.

Reply:
I see. IMHO I feel like the best course of actions is to revert #11153, also revert #12661 by me a few days ago, close this PR, and then open a ticket to move order by to the charts.